core/arch/arch.c

```diff
@@ -497,6 +497,8 @@ static void shared_gencode_init(IF_X86_64_ELSE(gencode_mode_t gencode_mode, void
     bool x86_to_x64_mode = false;
 #endif
 
+    pthread_jit_write();
```

Comment:
Not sure these calls are everywhere they should be, nor paired with the counterpart at all the right places.  I don't see a write call for building a new block?  Is that the vmcode_get_writable_addr()?  But that's called many times...probably better to call the write just once at a higher level?  How about some XXX i#5383 comments for each of these calls in this PR explaining that they may not be ideally placed or paired or exhaustive and an audit should be done to improve.